src/Compilers/CSharp/Portable/BoundTree/Formatting.cs

```diff
@@ -64,7 +65,26 @@ internal partial class BoundTupleExpression
     {
         public override object Display
         {
-            get { return (object)this.Type ?? "<tuple>"; }
+            get
+            {
+                var pooledBuilder = PooledStringBuilder.GetInstance();
+                var builder = pooledBuilder.Builder;
+                var arguments = this.Arguments;
+
+
+                builder.Append('(');
+                builder.Append(arguments[0].Display);
+
+                for(int i = 1; i < arguments.Length; i++)
+                {
+                    builder.Append(", ");
+                    builder.Append(arguments[i].Display);
```

Comment:
Would it make sense to add an AppendDisplay(StringBuilder) function so that nested cases could be handled more efficiently?